src/main/java/hudson/plugins/promoted_builds/parameters/PromotedBuildParameterDefinition.java

```diff
@@ -171,22 +173,34 @@ public FormValidation doCheckJobName(@AncestorInPath Item project, @QueryParamet
             project.checkPermission(Item.CONFIGURE);
 
             if (StringUtils.isNotBlank(value)) {
-                AbstractProject p = Jenkins.getInstance().getItem(value,project,AbstractProject.class);
-                if(p==null)
+                // JENKINS-25011: also look for jobs in folders.
+                AbstractProject p = Jenkins.getInstance().getItemByFullName(value, AbstractProject.class);
+                if (p==null) {
+                    // suggest full name so that getBuilds() can find item.
                     return FormValidation.error(hudson.tasks.Messages.BuildTrigger_NoSuchProject(value,
-                            AbstractProject.findNearest(value, project.getParent()).getRelativeNameFrom(project)));
+                            AbstractProject.findNearest(value, project.getParent()).getFullName()));
+                }
 
             }
 
             return FormValidation.ok();
         }
 
-        public AutoCompletionCandidates doAutoCompleteJobName(@QueryParameter String value) {
+        public AutoCompletionCandidates doAutoCompleteJobName(@AncestorInPath Item project, @QueryParameter String value) {
+            if (!project.hasPermission(Item.CONFIGURE)) {
+                return new AutoCompletionCandidates();
+            }
+            project.checkPermission(Item.CONFIGURE);
+
+            ItemGroup parent = project.getParent();
             AutoCompletionCandidates candidates = new AutoCompletionCandidates();
-            List<AbstractProject> jobs = Jenkins.getInstance().getItems(AbstractProject.class);
+
+            // JENKINS-25011: look for jobs in all folders.
+            List<AbstractProject> jobs = Jenkins.getInstance().getAllItems(AbstractProject.class);
```

Comment:
Minor: Jenkins.getInstance() may return null, but it's not realistic for the right usage of autocompletion handlers